llama-index-integrations/llms/llama-index-llms-mymagic/llama_index/llms/mymagic/base.py

```diff
@@ -0,0 +1,170 @@
+import time
+from typing import Any, Dict, Optional
+import httpx
+import asyncio
+import requests
+from llama_index.core.base.llms.types import (
+    CompletionResponse,
+    CompletionResponseGen,
+    ChatResponse,
+    ChatResponseGen,
+    ChatResponseAsyncGen,
+    CompletionResponseAsyncGen,
+    LLMMetadata,
+)
+from llama_index.core.llms.llm import LLM
+from llama_index.core.bridge.pydantic import Field
+
+
+class MyMagicAI(LLM):
+    base_url_template: str = "https://{model}.mymagic.ai"
+    api_key: str = None
+    model: str = Field(default="mistral7b", description="The MyMagicAI model to use.")
```

Comment:
Do users need to worry about prompt formatting, or is mymgaic handling this in their API?